runtime/src/topo/hwloc/topo-hwloc.c

```diff
@@ -575,8 +575,17 @@ static void partitionResources(void) {
   if (numLocalesOnNode > 1) {
     oversubscribed = true;
   }
-  if ((numColocales > 0) && (numLocalesOnNode <= numColocales)){
-    numPartitions = numColocales;
+  if (numColocales > 0) {
+    if (numLocalesOnNode <= numColocales) {
+      // There are fewer colocales than there should be, probably because the
+      // number of locales isn't evenly divisable by the number of nodes.
+      // Partition resources as if there are the full complement of
+      // colocales, but set the number of colocales to the actual number.
+      numPartitions = numColocales;
+      chpl_set_num_colocales_on_node(numLocalesOnNode);
```

Comment:
Should there be a warning in this case?

Reply:
No, currently it can only happen if the user launches the job manually, in which case all bets are off (i.e., we assume the user knows what they are doing). Throughout the `colocale` functionality I've implemented it to support L locales on N nodes, where N may not evenly divide L. I figure we may want that some day.